enhancements/console/samples.md

```diff
@@ -0,0 +1,396 @@
+
+Samples are no longer tied to the exact OpenShift version and could be updated with the operator. Also customers can add their own samples if needed.
+
+This will also allow us to migrate from the (deprecated) [samples-operator](https://github.com/openshift/cluster-samples-operator) to a simple custom resource in the future and to add other sample types in the future.
+
+### User Stories
+
+* As an OpenShift Serverless engineer, I want to provide additional samples that are tied to an operator version, not an OpenShift release.
+* As a cluster administrator, I want to add additional samples to the Developer Console sample catalog.
+
+### Goals
+
+1. This feature should allow other teams to provide their Developer Console samples without applying them to the sample-operator or the console-operator.
+2. Samples should support localization/translation.
+
+### Non-Goals
+
+1. We don't want to turn off the support for the (deprecated) [samples-operator](https://github.com/openshift/cluster-samples-operator) right now. The console will not provide any samples on its own (in 4.14), but it might be an option to migrate these samples in the future.
+2. A way to hide installed samples. Operators creating a sample resource should have a way to opt out of their samples.
+
+## Proposal
+
+### Workflow Description
+
+The Console Operator will provide a cluster-wide `ConsoleSample` CRD.
+
+Other operators like the OpenShift Serverless operator will install their own `ConsoleSample` CRs.
+
+The web console will automatically load all `ConsoleSample` CRs and show them in the sample catalog.
+
+Users can select one of samples and follow the import flows to create a demo application.
+
+### API Extensions
+
+The console operator automatically installs a new `ConsoleSample` CRD.
+
+### Risks and Mitigations
+
+#### The OpenShift Console is optional and could be disabled
+
+Other operators should expect that the `ConsoleSample` CRD is unavailable like the other `Console*` CRDs.
+
+### Drawbacks
+
+Another extra (but specific) custom resource just for console samples.
+
+## Design Details
+
+An "Import from Git" example of how we have it today as part of the Builder Images annotation:
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: java-sample
+spec:
+  title: Java
+  abstract: Build and run Java applications using Maven and OpenJDK.
+  description: |
+    # About this sample
+    Build and run Java applications using Maven and OpenJDK.
+    # Links
+    * [Sample repository](https://github.com/jboss-openshift/openshift-quickstarts)
+  icon: base64 encoded image
+  provider: Red Hat
+  badge: Source to Image
+  tags:
+  - java
+  - jboss
+  - openjdk
```

Comment:
It's not clear to me why we have both of these, what's the difference and what are they used for? Is there somewhere we can link to for the design of the samples to show what/how these are used?

Reply:
I changed the naming from `badge` to `type` and explained it hopefully better.

5. `type` is an optional display label to group multiple sample types. It can have a maximum length of **20** characters.

   Currently, the type is shown a badge on the sample card tile in the top right corner (see screenshot above).

   Recommendation is a **singular term** that groups samples like "Builder Image", "Devfile" or "Serverless Function".

Also updated the screenshot in the doc:

![image](https://github.com/openshift/enhancements/assets/139310/13d45a39-7e06-4707-9ea9-5430277ac019)

Reply:
Yeah that works well, thanks